cmplxfoil/CMPLXFOIL.py

```diff
@@ -0,0 +1,1120 @@
+            raise ValueError("xDvDot and xSDot cannot both be None")
+
+        if mode not in ["FD", "CS"]:
+            raise ValueError(f'Jacobian vector product mode "{mode}" invalid. Must be either "FD" or "CS"')
+
+        geoDVs = list(self.DVGeo.getValues().keys())
+        possibleDVs = self.possibleAeroDVs + geoDVs
+        for DV in xDvDot.keys():
+            if DV not in possibleDVs:
+                raise ValueError(f'Perturbed design variable "{DV}" is not valid')
+
+        if h is None:
+            if mode == "FD":
+                h = 1e-6
+            elif mode == "CS":
+                h = 1e-200j
+
+        if mode == "FD":
+            orig_funcs = deepcopy(self.funcs[self.curAP.name])
+
+        # Process the Xs perturbation
+        if xSDot is None:
+            xsdot = np.zeros_like(self.coords0)
+        else:
+            xsdot = xSDot
+
+        # For the geometric xDvDot perturbation we accumulate into the
+        # already existing (and possibly nonzero) xsdot and xvdot
+        geoxDvDot = {k: xDvDot[k] for k in geoDVs if k in xDvDot}
+        if xDvDot is not None and self.DVGeo is not None:
+            xsdot += self.DVGeo.totalSensitivityProd(geoxDvDot, self.curAP.ptSetName, config=self.curAP.name).reshape(
+                xsdot.shape
+            )
+
+        # Perturb the coordinates
+        orig_coords = self.getCoordinates()
+        if mode == "FD":
+            self.setCoordinates(orig_coords + xsdot * h)
+        else:
+            self.setCoordinatesComplex(orig_coords + xsdot * h)
+
+        orig_alpha = copy(self.curAP.alpha)
+        if "alpha" in xDvDot.keys():
+            self.curAP.alpha += xDvDot["alpha"] * h
+
+        self.__call__(self.curAP, useComplex=mode == "CS", deriv=True)
+
+        # Compute the Jacobian vector products
+        jacVecProd = {}
+        for f in self.functionList:
+            if mode == "FD":
+                jacVecProd[f] = (self.funcs[self.curAP.name][f] - orig_funcs[f]) / h
+            else:
+                jacVecProd[f] = np.imag(self.funcsComplex[self.curAP.name][f]) / np.imag(h)
+
+        # Reset the perturbed variables
+        self.curAP.alpha = orig_alpha
+        if mode == "FD":
+            self.setCoordinates(orig_coords)
+            self.funcs[self.curAP.name] = orig_funcs
+        else:
+            self.setCoordinatesComplex(orig_coords)
+
+        return jacVecProd
+
+    def getTriangulatedMeshSurface(self, offsetDist=1.0):
+        """
+        This function returns a pyGeo surface. The intent is
+        to use this for DVConstraints.
+
+        .. note::
+            This method requires the pyGeo library
+
+        Parameters
+        ----------
+        offsetDist : float
+            Distance to extrude airfoil (same units as airfoil coordinates)
+
+        Returns
+        -------
+        pyGeo surface
+            Extruded airfoil surface
+        """
+        from pygeo.pyGeo import pyGeo
```

Comment:
Should this be wrapped in a `try` / `except` to have a clearer error if someone doesn't have pyGeo installed?